packages/vue-pdf-viewer/src/components/PdfViewer.vue

```diff
@@ -126,15 +127,60 @@ export default {
 
             this.currentScale = newScale > MAX_SCALE ? MAX_SCALE : newScale
         },
-        onPagesLoaded() {
+        async onPagesLoaded() {
             this.pdfViewer.currentScaleValue = this.currentScale = this.defaultScale
 
             const event = new Event('PDFViewer:pagesLoaded')
             window.dispatchEvent(event)
 
             const scrollbarWidth = this.$refs.viewerContainer.offsetWidth - this.$refs.viewerContainer.clientWidth
             this.$refs.viewerControls.style.width = `calc(100% - ${scrollbarWidth}px)`
+
+            if (await this.pdfJsHelper.hasForm(this.pdfDocument) && this.pdfjsViewerOptions.annotationMode === 2) {
+              await this.prepareRequiredFormFields()
+            }
         },
+      async prepareRequiredFormFields() {
+          const requiredFormFields = await this.pdfJsHelper.getRequiredFormFields(this.pdfDocument)
+
+          if (requiredFormFields?.length > 0) {
+              requiredFormFields.forEach((formField) => {
+                  const htmlElement = document.querySelector(`[data-element-id="${formField.id}"]`)
+                  this.updateField(formField.id, formField.fieldName, htmlElement.type, htmlElement)
+
+                  htmlElement.addEventListener('input', async (event) => {
+                    this.updateField(formField.id, formField.fieldName, event.target.type, event.target)
+                    const formIsValid = await this.validateReactiveFormFields()
+                    this.$emit('required-fields-filled', formIsValid)
+                  })
+              })
+
+              const formIsValid = await this.validateReactiveFormFields()
+              this.$emit('required-fields-filled', formIsValid)
+          }
+      },
+      async updateField(fieldId, fieldName, fieldType, htmlElement) {
+          let fieldValue
+
+          switch(fieldType) {
+              case 'radio':
+              case 'checkbox':
+                fieldValue = htmlElement.checked
+                break
+              default:
+                fieldValue = htmlElement.value
+                break
+          }
```

Comment:
Do you have prettier correctly setup for this project? It doesn't seem to format it correctly.

Reply:
It wasn't running anymore for some reason and I did not notice it. A IDE restart solved it and all formatting should be good now